tests/testsuite/cargo_add/overwrite_git_with_path_base/mod.rs

```diff
@@ -0,0 +1,39 @@
+use cargo_test_support::compare::assert_ui;
+use cargo_test_support::current_dir;
+use cargo_test_support::file;
+use cargo_test_support::prelude::*;
+use cargo_test_support::str;
+use cargo_test_support::Project;
+
+#[cargo_test]
+fn case() {
+    cargo_test_support::registry::init();
+    for ver in [
+        "0.1.1+my-package",
+        "0.2.0+my-package",
+        "0.2.3+my-package",
+        "0.4.1+my-package",
+        "20.0.0+my-package",
+        "99999.0.0+my-package",
+        "99999.0.0-alpha.1+my-package",
+    ] {
+        cargo_test_support::registry::Package::new("cargo-list-test-fixture-dependency", ver)
+            .publish();
+    }
+
+    let project = Project::from_template(current_dir!().join("in"));
+    let project_root = project.root();
+    let cwd = project_root.join("primary");
+
+    snapbox::cmd::Command::cargo_ui()
+        .arg("add")
+        .arg_line("cargo-list-test-fixture-dependency --path dependency --base my_base")
```

Comment:
> tests/testsuite/cargo_add/overwrite_git_with_path_base/mod.rs

I don't think this test case is needed.  The important part is that any `--path` causes `git` fields to be wiped out.

Testing this would be like testing that `git` without any other fields and with `["rev", "tag", "branch"]` all successfully replace a `path`